File: docs-examples/codecs-java/src/test/java/io/micronaut/configuration/lettuce/docs/PooledRedisCommands.java
package io.micronaut.configuration.lettuce.docs;

import io.lettuce.core.api.StatefulRedisConnection;
import io.lettuce.core.support.AsyncPool;
import io.micronaut.context.annotation.Requires;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;

import java.util.concurrent.TimeUnit;

@Requires(property = "redis.pool.enabled", value = "true")
@Singleton
public final class PooledRedisCommands {

    // tag::pool[]
    @Inject AsyncPool<StatefulRedisConnection<String, String>> pool;

    void commands() throws Exception {
        StatefulRedisConnection<String, String> connection = pool.acquire().get(5, TimeUnit.SECONDS);
        try {
            connection.sync().set("first", "one");
        } finally {
            pool.release(connection).get(5, TimeUnit.SECONDS);
        }
    }
    // end::pool[]
}

File: docs-examples/codecs-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/PooledRedisCommands.kt
package io.micronaut.configuration.lettuce.docs

import io.lettuce.core.api.StatefulRedisConnection
import io.lettuce.core.support.AsyncPool
import io.micronaut.context.annotation.Requires
import jakarta.inject.Inject
import jakarta.inject.Singleton
import java.util.concurrent.TimeUnit

@Requires(property = "redis.pool.enabled", value = "true")
@Singleton
class PooledRedisCommands {

    // tag::pool[]
    @Inject
    lateinit var pool: AsyncPool<StatefulRedisConnection<String, String>>

    fun commands() {
        val connection = pool.acquire().get(5, TimeUnit.SECONDS)
        try {
            connection.sync().set("first", "one")
        } finally {
            pool.release(connection).get(5, TimeUnit.SECONDS)
        }
    }
    // end::pool[]
}
